src/pages/resultsView/survival/SurvivalChart.tsx

```diff
@@ -847,45 +872,141 @@ export default class SurvivalChartExtended
         );
         return point;
     }
-    @computed get numberOfSamplesAtRisk() {
-        var timePoints = scaleLinear()
-            .domain([0, this.sliderValue])
-            .ticks(18);
+    @observable userDefinedTimePoints: boolean = false;
+
+    @computed get timePointsForNumberAtRiskLabels() {
+        // See if the user defined an input
+        if (this.userDefinedTimePoints) {
+            return [10, 30, 80];
+        } else {
+            return scaleLinear()
+                .domain([0, this.sliderValue])
+                .ticks(18);
+        }
+    }
 
-        const numberAtRisk = _.groupBy(
-            this.calculateGroupSize(timePoints.map(item => item)),
+    @action.bound
+    calculateLabelWidth(text: number, fontFamily: string, fontSize: number) {
```

Comment:
We should not mark this function as an `action`, because it does not modify any observable values. Also, we can probably use the `getTextWidth` function instead of writing a new one.
https://github.com/cBioPortal/cbioportal-frontend/blob/0bcf4809624f0dc7bd5d9169c2bef3172281b57c/packages/cbioportal-frontend-commons/src/lib/TextTruncationUtils.ts#L17

Reply:
I moved `calculateLabelWidth` to the `SurvivalUtils ` so it becomes a function.. I checked and there is a difference in the width if we calculate it for a label or just the plain text. I believe the `getTextWidth `function would still allow for overlap in the survival chart.